frontend/src/views/WheresMyGeneV2/components/InfoPanel/components/SourceData/connect.ts

```diff
@@ -0,0 +1,41 @@
+import { useCallback, useContext, useMemo, useState } from "react";
+import { StateContext } from "src/views/WheresMyGeneV2/common/store";
+import {
+  aggregateCollectionsFromDatasets,
+  useFilterDimensions,
+} from "src/common/queries/wheresMyGene";
+import { Collections } from "./types";
+import { track } from "src/common/analytics";
+import { EVENTS } from "src/common/analytics/events";
+import { SOURCE_DATA } from "./constants";
+
+export const useConnect = () => {
+  const [hoverStartTime, setHoverStartTime] = useState(0);
+  const useHandleHoverEnd = (event: EVENTS, payload = {}) => {
+    return useCallback(() => {
+      if (Date.now() - hoverStartTime > 2 * 1000) {
+        track(event, payload);
+      }
+    }, [event, payload]);
+  };
+
+  const handleBadgeHoverEnd = useHandleHoverEnd(
+    EVENTS.WMG_LEGEND_QUESTION_BUTTON_HOVER,
+    { label: SOURCE_DATA }
+  );
+
+  const { selectedFilters } = useContext(StateContext);
+  const { data: filterDimensions } = useFilterDimensions();
+
+  let { datasets = [] } = filterDimensions;
+  if (selectedFilters.datasets.length > 0) {
+    datasets = datasets.filter((dataset) =>
+      selectedFilters.datasets.includes(dataset.id)
+    );
+  }
```

Comment:
Sorry I still need help understanding this piece of code! What is it doing and why is it needed? Thank you!

Just looking at the code, I'm a little confused because we've removed `datasets` from the left panel, so intuitively I don't know what the value of `selectedFilters.datasets` can be beside `[]` all the time, because users can't select datasets anymore 🤔 ?

Reply:
Not sure why this happened again. But this comment hasn't been addressed but somehow it's resolved. Reopening!